src/textual/geometry.py

```diff
@@ -127,6 +141,34 @@ def __contains__(self, other: Any) -> bool:
         width, height = self
         return width > x >= 0 and height > y >= 0
 
+    
+C = TypeVar("C", bound="Corners")
+
+
+class Corners(NamedTuple):
```

Comment:
This probably isn't worth doing right now. There may never be a need to refer to the contents by name. 

You'll see expressions like this in the code:

```python
x, y, x2, y2 = self.corners
```

Unpacking to locals like that is considerably faster than attribute lookup. In the case of `corners` it is exclusively used like this.

This *may* change in the future, but for now I think its best left as a 4xtuple.